src/items.rs

```diff
@@ -1892,25 +1914,29 @@ pub(crate) fn rewrite_struct_field(
     if prefix.is_empty() && !attrs_str.is_empty() && attrs_extendable && spacing.is_empty() {
         spacing.push(' ');
     }
+
     let orig_ty = shape
         .offset_left(overhead + spacing.len())
-        .and_then(|ty_shape| field.ty.rewrite(context, ty_shape));
+        .and_then(|ty_shape| field.ty.rewrite_result(context, ty_shape).ok());
+
     if let Some(ref ty) = orig_ty {
         if !ty.contains('\n') && !contains_comment(context.snippet(missing_span)) {
-            return Some(attr_prefix + &spacing + ty);
+            return Ok(attr_prefix + &spacing + ty);
         }
     }
```

Comment:
Since we are gonna continue formatting though `offset_left` returns None, I replaced `ty.rewrite` with `ty.rewrite_result.ok()` instead of converting Option into RwResult like below. 
```rust
let orig_ty = shape
        .offset_left(overhead + spacing.len())
        .max_width(shape.width, field.span())
        .and_then(|ty_shape| field.ty.rewrite_result(context, ty_shape));

    if let Ok(ref ty) = orig_ty {
        if !ty.contains('\n') && !contains_comment(context.snippet(missing_span)) {
            return Ok(attr_prefix + &spacing + ty);
        }
    }
```
Although it seems somewhat pointless to calling rewrite_result.ok() instead of rewrite, I did so since rewrite will return Result after all. 
It's pretty minor issue, but I want to hear your suggestion if you have one.

Reply:
I think calling `rewrite_result.ok()` is fine. We could also follow your suggestion, but I don't think it makes much of a difference right now.